### PR TITLE
Fix pathfinder nodes not overriding equals and hashCode

### PR DESCRIPTION
**Pull request by fullwall**

### CODE REVIEW

**Comment by windy1** [comment]
If you could add some unit tests that would be supreme. We really need unit tests for the AI package.

File: api/src/main/java/org/spout/api/ai/AStarNode.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * This file is part of Spout.
  *
- * Copyright (c) 2011 Spout LLC <http://www.spout.org/>
+ * Copyright (c) 2011 Spout LLC <http://www.spout.org/> <http://www.spout.org/>
```
Comment (kitskub) [inline api/src/main/java/org/spout/api/ai/AStarNode.java]: Hmmmm

**Comment by kitskub** [comment]
Are you breaking or fixing the formatting?

**Comment by fullwall** [comment]
@W1ndwaker - I can work on some unit tests.

File: api/src/main/java/org/spout/api/ai/AStarGoal.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 public interface AStarGoal<T extends AStarNode> {
 	/**
 	 * Returns the cost of moving between the two supplied {@link AStarNode}s.
-	 *
+	 *
```
Comment (kitskub) [inline api/src/main/java/org/spout/api/ai/AStarGoal.java]: These changes are not right.

File: api/src/main/java/org/spout/api/ai/AStarMachine.java [hunk 3]
```diff
@@ -55,7 +55,7 @@ private AStarStorage getInitialisedStorage(AStarGoal<N> goal, N start) {
 
 	/**
 	 * Creates an {@link AStarState} that can be reused across multiple invocations of {{@link #run(AStarState, int)}.
-	 *
+	 *
```
Comment (kitskub) [inline api/src/main/java/org/spout/api/ai/AStarMachine.java]: See above.

File: api/src/main/java/org/spout/api/ai/AStarMachine.java [hunk 4]
```diff
@@ -127,8 +127,9 @@ public P runFully(AStarGoal<N> goal, N start) {
 	}
 
 	/**
-	 * Runs the machine fully until the iteration limit has been exceeded. This will use the supplied goal and start to generate neighbours until the goal state has been reached using the A* algorithm.
-	 *
+	 * Runs the machine fully until the iteration limit has been exceeded. This will use the supplied goal and start to generate neighbours until the goal state has been reached using the A*
+	 * algorithm.
```
Comment (kitskub) [inline api/src/main/java/org/spout/api/ai/AStarMachine.java]: Nope.

**Comment by kitskub** [comment]
@fullwall can you rebase and update for my comments?

**Comment by fullwall** [comment]
@kitskub - I think I messed up the history with the rebase...

**Comment by kitskub** [comment]
Ooh uh. I pull this locally and commit it.

**Comment by kitskub** [comment]
Pulled. Thanks!

**Pull request closed without merging**